plugins/out_stackdriver/stackdriver.c

```diff
@@ -2491,6 +2491,16 @@ static void cb_stackdriver_flush(struct flb_event_chunk *event_chunk,
     if (ret == 0) {
         update_http_metrics(ctx, event_chunk, ts, c->resp.status);
     }
+
+    /* Update retry count if necessary */
+    if (ret_code == FLB_RETRY) {
+        char tmp[32];
+
+        /* convert status to string format */
+        snprintf(tmp, sizeof(tmp) - 1, "%i", c->resp.status);
```

Comment:
I think we should use [flb_sds](https://github.com/fluent/fluent-bit/blob/master/DEVELOPER_GUIDE.md#strings) here instead, and use [flb_sds_printf](https://github.com/fluent/fluent-bit/blob/master/src/flb_sds.c#L408). See this [function from the Azure Blob plugin](https://github.com/fluent/fluent-bit/blob/master/plugins/out_azure_blob/azure_blob_appendblob.c#L37) that shows an example of what I mean.

Reply:
The label keys for the metrics are all `char *` and this is what we do for all our other metrics as well so kept it the same for consistency.

Reply:
`flb_sds_t` is a typedef on `char *` so it can be used that way. If all our metrics work this way though then we can leave it for this PR. I think we should refactor this sometime soon though.